### PR TITLE
Examples compile again

### LINKED ISSUE #82: C++ Templates broken
The Sample, Iterative, and Command Templates all fail to compile.

### PR DESCRIPTION
**Pull request by calcmogul**
The C++ examples were also modified to use the frc namespace.

Closes #82

### CODE REVIEW

**Comment by sciencewhiz** [comment]
fixes #82

File: edu.wpi.first.wpilib.plugins.cpp/resources/templates/command-based/Robot.cpp [hunk 1]
```diff
@@ -39,27 +42,28 @@ class Robot : public IterativeRobot {
...
-		if (autoSelected == "My Auto") {
-			autonomousCommand.reset(new MyAutoCommand());
-		}
-		else {
-			autonomousCommand.reset(new ExampleCommand());
-		} */
+	void AutonomousInit() override {
+		/* std::string autoSelected = frc::SmartDashboard::GetString("Auto Selector", "Default");
+		 if (autoSelected == "My Auto") {
+		 autonomousCommand.reset(new MyAutoCommand());
```
Comment (sciencewhiz) [inline edu.wpi.first.wpilib.plugins.cpp/resources/templates/command-based/Robot.cpp]: Should indent, even though it's inside the comment.

File: edu.wpi.first.wpilib.plugins.cpp/resources/templates/iterative/Robot.cpp [hunk 2]
```diff
@@ -2,16 +2,19 @@
...
 
-class Robot: public IterativeRobot {
+class Robot: public frc::IterativeRobot {
 public:
 	void RobotInit() {
-		chooser->AddDefault(autoNameDefault, static_cast<void*>(&autoNameDefault));
-		chooser->AddObject(autoNameCustom, static_cast<void*>(&autoNameCustom));
-		SmartDashboard::PutData("Auto Modes", chooser);
+		chooser.AddDefault(autoNameDefault,
+				const_cast<void*>(static_cast<const void*>(&autoNameDefault)));
```
Comment (sciencewhiz) [inline edu.wpi.first.wpilib.plugins.cpp/resources/templates/iterative/Robot.cpp]: Is there any way to simplify this?

File: edu.wpi.first.wpilib.plugins.java/resources/templates/examples/CANTalonPID/src/org/usfirst/frc/team190/robot/Robot.java [hunk 3]
```diff
@@ -12,51 +11,56 @@
...
+		// be set, but are ignored here).
+		// These must all be positive floating point numbers (reverseSensor will
+		// multiply the sensor values by negative one in case your sensor is
+		// flipped
+		// relative to your motor).
+		// These values are in units of throttle / sensor_units where throttle
+		// ranges
+		// from -1023 to +1023 and sensor units are from 0 - 1023 for analog
+		// potentiometers, encoder ticks for encoders, and position / 10ms for
+		// speeds.
```
Comment (sciencewhiz) [inline edu.wpi.first.wpilib.plugins.java/resources/templates/examples/CANTalonPID/src/org/usfirst/frc/team190/robot/Robot.java]: Text shold be wrapped better

File: edu.wpi.first.wpilib.plugins.cpp/resources/templates/iterative/Robot.cpp (hunk 2 above)
Comment (calcmogul) [reply edu.wpi.first.wpilib.plugins.cpp/resources/templates/iterative/Robot.cpp]: Not really. The static_cast could be replaced with a C-style cast (which is bad practice), but the const_cast is required. allwpilib#433 would be a better way to handle it.

**Pull request merged**